src/viewer.js

```diff
@@ -3137,6 +3137,10 @@ function onCanvasKeyDown( event ) {
 
     this.raiseEvent('canvas-key', canvasKeyDownEventArgs);
 
+    if (!canvasKeyDownEventArgs.preventDefaultAction) {
+        this._keysDown[canvasKeyDownEventArgs.originalEvent.code] = true; // Mark this key as held down in the viewer's internal tracking object
```

Comment:
This should probably be inside the if statement below; that way it is not engaged when those modifier keys are held down, maintaining the logic.